include/swift/AST/AnyFunctionRef.h

```diff
@@ -137,6 +137,26 @@ class AnyFunctionRef {
     return CE->getType()->castTo<FunctionType>()->isNoEscape();
   }
 
+  bool isObjC() const {
+    if (auto afd = TheFunction.dyn_cast<AbstractFunctionDecl *>()) {
+      return afd->isObjC();
+    }
+    if (TheFunction.dyn_cast<AbstractClosureExpr *>()) {
+      // Closures are never @objc.
+      return false;
+    }
+    llvm_unreachable("unexpected AnyFunctionRef representation");
```

Comment:
Why not `assert(TheFunction.is<AbstractClosureExpr *>())`? I think this is more readable.